### PR TITLE
MomConnect add metrics

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**

### CODE REVIEW

File: src/ussd_clinic.js [hunk 1]
```diff
@@ -39,10 +41,95 @@ go.app = function() {
...
+            self.metric_prefix = [self.env, self.im.config.name].join('.');
+            self.store_name = [self.env, self.im.config.name].join('.');
+
+            self.attach_session_length_helper(self.im);
+
+            mh = new MetricsHelper(self.im);
+            mh
+                // Total unique users
+                // This adds <env>.servicerating.sum.unique_users 'last' metric
+                // As well as <env>.servicerating.sum.unique_users.transient 'sum' metric
```
Comment (gsvr) [inline src/ussd_clinic.js]: `servicerating`?

File: src/ussd_clinic.js [hunk 2]
```diff
@@ -39,10 +41,95 @@ go.app = function() {
...
+
+            self.im.on('session:new', function(e) {
+                return self
+                .incr_kv(self.im, [self.store_name, 'sessions'].join('.'))
+                .then(function() {
+                    self.im.metrics.fire.inc([self.env, 'sum', 'sessions'].join('.'));
+                });
+            });
+
+            self.im.user.on('user:new', function(e) {
```
Comment (gsvr) [inline src/ussd_clinic.js]: Did we not say we'd do this and sessions above with the Metrics Helper?

File: src/ussd_clinic.js [hunk 3]
```diff
@@ -228,7 +315,15 @@ go.app = function() {
                     new Choice('state_start', $('Start new registration'))
                 ],
                 next: function(choice) {
-                    return choice.value;
+                    if (choice.value !== 'state_start') {
+                        return self
+                        .fire_complete(self.im.state.name, 1)
```
Comment (gsvr) [inline src/ussd_clinic.js]: Won't this fire twice for `state_timed_out` now?

File: test/ussd_clinic.test.js [hunk 4]
```diff
@@ -61,6 +70,294 @@ describe("app", function() {
...
+                            , '5'  // state_due_date_month
+                        )
+                        .check(function(api) {
+                            var metrics = api.metrics.stores.test_metric_store;
+                            assert.deepEqual(metrics['test.ussd_clinic.state_start.no_complete'].values, [1]);
+                            assert.deepEqual(metrics['test.ussd_clinic.state_start.no_complete.transient'].values, [1]);
+                            assert.deepEqual(metrics['test.ussd_clinic.state_clinic_code.no_complete'].values, [1]);
+                            assert.deepEqual(metrics['test.ussd_clinic.state_clinic_code.no_complete.transient'].values, [1]);
+                            assert.deepEqual(metrics['test.ussd_clinic.state_timed_out.no_complete'].values, [1,2]);
+                            assert.deepEqual(metrics['test.ussd_clinic.state_timed_out.no_complete.transient'].values, [1,1]);
```
Comment (gsvr) [inline test/ussd_clinic.test.js]: this shouldn't fire twice

**Comment by NicolaasDieBaas** [comment]
@gsvr please review

**Review by gsvr — APPROVED** [review]
:+1:

**Pull request merged**